src/controllers/authenticationController.ts

```diff
@@ -105,6 +108,80 @@ export default class AuthenticationController {
       .send(metadata);
   }
 
+  /**
+   * Returns data about the current user session.
+   */
+  public async getSessionState(
+    req: express.Request
+  ): Promise<Either<Error, ISessionState>> {
+    try {
+      const maybeToken = await this.extractTokenFromRequest(req);
+      if (isNone(maybeToken)) {
+        return left(new Error("No token in the request"));
+      }
+
+      const errorOrSession = await this.sessionStorage.get(maybeToken.value);
+      if (isLeft(errorOrSession)) {
+        const errorOrSessionState = await this.refreshUserToken(
+          maybeToken.value
+        );
+
+        if (isLeft(errorOrSessionState)) {
+          const error = errorOrSessionState.value;
+          return left(error);
+        }
+
+        const sessionState = errorOrSessionState.value;
+        return right(sessionState);
+      }
+
+      const session = errorOrSession.value;
+      const publicSession = {
+        expireAt: session.expireAt,
+        expired: session.expired
+      };
+      return right(publicSession);
+    } catch (err) {
+      return left(new Error("Error while loading the session"));
+    }
+  }
+
+  /**
+   * Extracts the Bearer token from the authorization header in the request.
+   */
+  private async extractTokenFromRequest(
```

Comment:
this function contains no async code, so there's no need to be labeled async and return a Promise